pkg/services/baremetal/remediation/remediation.go

```diff
@@ -59,9 +59,7 @@ func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err erro
 			if err := s.setOwnerRemediatedConditionNew(ctx); err != nil {
 				err = fmt.Errorf("failed to set remediated condition on capi machine: %w", err)
 				record.Warn(s.scope.BareMetalRemediation, "FailedSettingConditionOnMachine", err.Error())
-				return res, err
 			}
-			return res, err
```

Comment:
I think there was a misunderstanding.. I said do not return an error, not do not return at all :D Please keep the return here!